primitives/core/src/bounded/bounded_btree_map.rs

```diff
@@ -328,6 +333,15 @@ where
 	}
 }
 
+impl<K, V, S> DerefMut for BoundedBTreeMap<K, V, S>
+where
+	K: Ord,
+{
+	fn deref_mut(&mut self) -> &mut Self::Target {
+		&mut self.0
+	}
+}
+
```

Comment:
no, we don't want `DerefMut` on `BoundedBTreeMap` because it opens it up to misuse.
This allows all inner `BTreeMap` api be exposed, and the problem are all the _addition/insertion_ methods on `BTreeMap` which can now be called directly with no bounds check.

Reply:
Alright. I have gone ahead to impelement your suggestions.
But can we still opt for the private struct map you suggested here https://github.com/paritytech/substrate/pull/12562#discussion_r1030580412

Or the PR is good as is?

Reply:
I think PR is good as is